src/components/Analyze/AnalyzeContext.tsx

```diff
@@ -34,6 +43,15 @@ export const AnalyzeProvider = ({
 		(state) => state.consumptionCombinedData,
 	)
 
+	const doughnutChartData: ChartDataType[] = consumptionCombinedData.map(
+		(each, index) => ({
+			id: each.category,
+			label: each.category,
+			value: each.price,
+			color: chartColors[index % 4],
```

Comment:
```suggestion
			color: chartColors[index % chartColors.length],
```
이렇게 하면 좀 더 유연해질 것 같네요.